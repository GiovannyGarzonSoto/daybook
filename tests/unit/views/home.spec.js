import Home from '@/views/Home'
import { shallowMount } from '@vue/test-utils'

describe('Home view tests', () => {
    it('should be render the component correctly', () => {
        const wrapper = shallowMount(Home)
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('click in a button should be redirect to no-entry', () => {
        const mockRouter = {
            push: jest.fn()
        }
        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })
        wrapper.find('button').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({name: 'no-entry'})
    })
})