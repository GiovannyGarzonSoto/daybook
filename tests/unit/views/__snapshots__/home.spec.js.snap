// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view tests should be render the component correctly 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola mundo</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">secondary</button><button class="btn btn-success">success</button>
</div>
`;
